business/checkers/destinationrules/no_dest_checker.go

```diff
@@ -48,10 +50,16 @@ func (n NoDestinationChecker) Check() ([]*models.IstioCheck, bool) {
 										}
 									}
 									if !n.hasMatchingWorkload(fqdn.Service, stringLabels) {
-										validation := models.Build("destinationrules.nodest.subsetlabels",
-											"spec/subsets["+strconv.Itoa(i)+"]")
-										validations = append(validations, &validation)
-										valid = false
+										if n.isSubsetReferenced(dHost, innerSubset["name"].(string)) {
+											validation := models.Build("destinationrules.nodest.subsetlabels",
+												"spec/subsets["+strconv.Itoa(i)+"]")
+											validations = append(validations, &validation)
+											valid = false
+										} else {
+											validation := models.Build("destinationrules.nodest.subsetlabelsnoterferenced",
```

Comment:
I see that you are using two different validations for this. If you want to do so, then in the documentation you should add one more entry covering this validation (KIA0210). Otherwise people will put this on google and won't found any documentation.

However, if you want to save this step, you can just change the severity for the existing validation
```golang
validation.Severity = models.Unknown
```

This should just keep the same validation but different severity. I am fine with both versions. Pick the one it works the best to your sense. Both have pros and cons we can evolve along the way.

Reply:
thank you @xeviknal I rethink and decided to use the same error code but different severity, this will cause less complications and confusions